src/app/beer_garden/db/mongo/pruner.py

```diff
@@ -30,7 +43,24 @@ def add_task(
             }
         )
 
+    def _cancel_outstanding(self):
+        """
+        Helper function for run to mark requests still outstanding after a certain
+        amount of time as canceled.
+        """
+        if isinstance(self._cancel_threshold, int):
```

Comment:
There's something a little bit awkward about having this check here.  I think there are a few options that would feel more natural.  An easy one might just be to set `_cance_outstanding` to '-1' if it's provided as `None`.  So in the `__init__`, instead of:

```python
self._cancel_threshold = cancel_threshold
```
you would do

```python
self._cancel_threshold = cancel_threshold or -1
```
Then you can skip the ` isinstance(self._cancel_threshold, int)` check entirely.

Another super minor thing, but I'd also move the `self._cancel_threshold > 0` check to where you are making the call to this function, rather than inside of it.  The reason for that being that someone reading or stepping through the code has to jump into this function to figure out that it's a no-op, as where if you do the check before the function call, it's clear right up front.